Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -109,28 +113,29 @@ - (void)validStringInserted:(NSString *)string
 }
 
 - (void)characterTyped:(unichar)c asInsertedCharacter:(BOOL)inserted previousCharacter:(unichar)previousCharacter {
-    NSCharacterSet *whitespaceSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
     __strong __auto_type delegate = self.delegate;
     // Determine the type of the character
     enum CharacterType {
-        CharacterTypeWhitespace = 0,
+        CharacterTypeSeparator = 0,
         CharacterTypeControlCharacter,
         CharacterTypeNormal
     };
     enum CharacterType currentCharacterType = CharacterTypeNormal;
-    if ([whitespaceSet characterIsMember:c]) {
-        currentCharacterType = CharacterTypeWhitespace;
+    if ([HKWMentionsStartDetectionStateMachine.whitespaceSet characterIsMember:c]) {
+        currentCharacterType = CharacterTypeSeparator;
     }
     else {
         // Get the control character set and see if the typed character is a control character
         NSCharacterSet *controlCharacterSet = [delegate controlCharacterSet];
         if (controlCharacterSet && [controlCharacterSet characterIsMember:c]) {
             currentCharacterType = CharacterTypeControlCharacter;
+        } else if ([HKWMentionsStartDetectionStateMachine.punctuationSet characterIsMember:c]) {
```

Comment:
you should just include this in the if above on line 124

Reply:
Control characters can overlap with punctuation character set which is why we are first checking for white space, then control characters and then punctuation.
For example, '@' is a punctuation character and can also be a control character.